src/WidgetBase.ts

```diff
@@ -436,6 +445,7 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		this._renderState = WidgetRenderState.RENDER;
 		if (this._dirty === true || this._cachedVNode === undefined) {
 			this._dirty = false;
+			this._nodeHandler.clear();
```

Comment:
I still think that this is in the wrong place, it will clear the map before running the renders and therefore always return back the default dimensions! This is were integration tests for using the meta with `WidgetBase` come in handy.

Reply:
moved down